source/src/protocols/cyclic_peptide/crosslinker/Lanthionine_Helper.cc

```diff
@@ -0,0 +1,311 @@
+//////////////////////
+
+/// @brief Given a pose and a selection of residues, add the linker,
+/// align it crudely to the selected residues, and set up covalent bonds.
+/// @details Must be defined by derived classes.  Version for asymmetric poses.
+void
+Lanthionine_Helper::add_linker_asymmetric(
+	core::pose::Pose & pose,
+	core::select::residue_selector::ResidueSubset const & selection
+) const {
+	//This will not compile until this is implemented.
+	//TODO
+	core::Size firstres, secondres; //Firstres has the backbone linkage, secondres has the sidechain linkage. - change var name to dala and cystine
+	get_lanthionine_residues_from_selection( selection, pose, firstres, secondres );
+	set_up_lanthionine_variants( pose, firstres, secondres ); //Defined in utility file, lanthionine_util.hh/cc.
+	//TR << "Added lanthionine variant types." << std::endl;
+	add_linker_bonds_asymmetric( pose, utility::vector1< core::Size >{ firstres, secondres }, 0 );
+}
+
+/// @brief Given a pose and a linker, add bonds between the linker and the residues
+/// that coordinate the linker.
+/// @details Can be called by add_linker_asymmetric().  Must be defined by derived
+/// classes (pure virtual).  Version for asymmetric poses.
+void
+Lanthionine_Helper::add_linker_bonds_asymmetric(
+	core::pose::Pose & pose,
+	utility::vector1< core::Size > const & res_indices,
+	core::Size const /*linker_index*/
+) const {
+	//This will not compile until this is implemented.
+	std::string const errmsg( "Error in Lanthionine_Helper::add_linker_bonds_asymmetric(): " );
+	protocols::simple_moves::DeclareBond declbond;
+	runtime_assert_string_msg( res_indices.size() == 2 && res_indices[1] != res_indices[2], errmsg + "Exactly two residues must be selected to set up linker bonds." );
+	runtime_assert_string_msg( res_indices[1] > 0 && res_indices[1] <= pose.total_residue(), errmsg + "The first residue index must be within the range of pose residues (1-" + std::to_string(pose.total_residue()) + "), but got " + std::to_string(res_indices[1]) + " instead." );
+	runtime_assert_string_msg( res_indices[2] > 0 && res_indices[2] <= pose.total_residue(), errmsg + "The second residue index must be within the range of pose residues (1-" + std::to_string(pose.total_residue()) + "), but got " + std::to_string(res_indices[2]) + " instead." );
+	set_up_lanthionine_bond_mover( declbond, pose, res_indices[1], res_indices[2] );
+	declbond.apply(pose);
+	correct_lanthionine_virtuals( pose, res_indices[1], res_indices[2] );
+	TR << "Added lanthionine bond." << std::endl;
+}
+
+/// @brief Given a pose and a selection of residues, add the linker,
+/// align it crudely to the selected residues, and set up covalent bonds.
+/// @details Must be defined by derived classes.  Version for symmetric poses.
+void
+Lanthionine_Helper::add_linker_symmetric(
+	core::pose::Pose & pose,
+	core::select::residue_selector::ResidueSubset const & selection
+) const {
+	//This will not compile until this is implemented.
```

Comment:
Is this comment still valid? (If not, I suggest removing it and the other similar comments.)

Reply:
Done